lib/HttpClient/CurlClient.php

```diff
@@ -47,12 +47,21 @@ public function request($method, $absUrl, $headers, $params, $hasFile)
             throw new Error\Api("Unrecognized method $method");
         }
 
+        // Create a callback to capture HTTP headers for the response
+        $rheaders = array();
+        $headerCallback = function ($curl, $header_line) use (&$rheaders) {
+          list($key, $value) = explode(":", trim($header_line), 1);
+          $rheaders[trim($key)] = trim($value);
+          return strlen($header_line);
+        };
+
         $absUrl = Util\Util::utf8($absUrl);
         $opts[CURLOPT_URL] = $absUrl;
         $opts[CURLOPT_RETURNTRANSFER] = true;
         $opts[CURLOPT_CONNECTTIMEOUT] = 30;
         $opts[CURLOPT_TIMEOUT] = 80;
         $opts[CURLOPT_RETURNTRANSFER] = true;
+        $opts[CURLOPT_HEADERFUNCTION] = $headerCallback;
```

Comment:
I originally split on `\r\n\r\n`, but ran into some strange errors with FileUploads. I think this approach works better because I don't need to munge the response at all. Also, from further down in the SO post: `it won't be affected by additional headers from proxies` which seems like a win as well.